smack-core/src/main/java/org/jivesoftware/smack/remoteEndpoint/HttpLookUp.java

```diff
@@ -0,0 +1,197 @@
+import org.jivesoftware.smack.util.rce.RemoteConnectionEndpoint;
+import org.jivesoftware.smack.xml.XmlPullParser;
+import org.jivesoftware.smack.xml.XmlPullParserException;
+
+import org.minidns.util.InetAddressUtil;
+
+/**
+ * The HTTP lookup method uses web host metadata to list the URIs of alternative connection methods.<br>
+ * <br>
+ * To obtain endpoints for Bosh or Websocket connections, use {@link RelAttribute#BOSH} and <br>
+ * {@link RelAttribute#WEBSOCKET} respectively with {@link #getEndpoints(RelAttribute)}. <br>
+ * <br>
+ * Example:<br>
+ * <pre>
+ * {@code
+ * HttpLookUp lookup = new HttpLookUp(host,usingHttps);
+ * List<RemoteConnectionEndpoint> endpoints = lookup.getEndpoints(RelAttribute.WEBSOCKET);
+ * }
+ * </pre>
+ * @see <a href="https://xmpp.org/extensions/xep-0156.html#http">
+*     HTTP Lookup Method.
+*     </a>
+ */
+public final class HttpLookUp {
+
+    private static final int BUFFER_SIZE = 256;
+    private static final String XRD_NAMESPACE = "http://docs.oasis-open.org/ns/xri/xrd-1.0";
+
+    /**
+     * Specifies a link relation for the selected type of connection.
+     */
+    enum RelAttribute {
+        /**
+         * Selects link relation attribute as "urn:xmpp:alt-connections:xbosh".
+         */
+        BOSH("urn:xmpp:alt-connections:xbosh"),
+        /**
+         * Selects link relation attribute as "urn:xmpp:alt-connections:websocket".
+         */
+        WEBSOCKET("urn:xmpp:alt-connections:websocket");
+        private final String attribute;
+        RelAttribute(String relAttribute) {
+            this.attribute = relAttribute;
+        }
+    }
+
+    private final int content_length;
+    private final HttpURLConnection httpConnection;
+
+    String obtainedXml = "";
+
+    /**
+     * Constructs a HTTP connection with the server and retrieves complete host metadata.
+     *
+     * @param host host that needs to be connected
+     * @param usingHttps true if secure HTTP being used, false otherwise
+     * @throws IOException exception due to input/output operations
+     */
+    public HttpLookUp(String host, boolean usingHttps) throws IOException {
+        final String metadataUrl = (usingHttps ? "https://" : "http://") + host + "/.well-known/host-meta";
+        httpConnection = getHttpConnection(metadataUrl);
+        content_length = httpConnection.getContentLength();
+
+        InputStream inputStream = httpConnection.getInputStream();
+        int remainingBytes = content_length;
+
+        // Keep reading from the input stream until read completely
+        while (remainingBytes > 0) {
+            byte[] byteBuffer = new byte[BUFFER_SIZE];
+            int bytes_read = inputStream.read(byteBuffer);
+            obtainedXml += new String(byteBuffer, 0, bytes_read, Charset.defaultCharset());
+            remainingBytes = remainingBytes - bytes_read;
+        }
+    }
+
+    private static HttpURLConnection getHttpConnection(String endpoint) throws IOException {
```

Comment:
You are talking of an endpoint as the String argument here, but the whole point of this class is to retrieve endpoints, so the naming of the argument is not ideal :P